cmd/oras/internal/display/metadata/interface.go

```diff
@@ -78,9 +79,25 @@ type ManifestPushHandler interface {
 	TaggedHandler
 }
 
+type manifestPackHandler interface {
+	OnIndexPacked(desc ocispec.Descriptor) error
+	OnIndexPushed(path string) error
+	OnCompleted(desc ocispec.Descriptor) error
+}
+
 // ManifestIndexCreateHandler handles metadata output for index create events.
 type ManifestIndexCreateHandler interface {
 	TaggedHandler
+	manifestPackHandler
+}
+
+// ManifestIndexUpdateHandler handles metadata output for index update events.
+type ManifestIndexUpdateHandler interface {
+	TaggedHandler
+	manifestPackHandler
+	OnManifestRemoved(digest digest.Digest) error
+	OnManifestAdded(manifestRef string, digest digest.Digest) error
+	OnIndexMerged(indexRef string, digest digest.Digest) error
 }
```

Comment:
Change digest to descriptor since it will be used for formatted output in the future.

Reply:
The parameter of `OnManifestRemoved(digest digest.Digest) error` must be a digest, since the descriptor is unknown. Changed the parameter of the other two functions as descriptors.